ood-portal-generator/lib/ood_portal_generator/hash_extensions.rb

```diff
@@ -0,0 +1,79 @@
+module OodPortalGenerator
+  module HashExtensions
```

Comment:
Can you put a note here that this is copied from rails and the license is ../../../RAILS-LICENSE

Reply:
I copied the text from https://github.com/OSC/ood_core/pull/329